src/http_utils.cpp

```diff
@@ -216,6 +218,22 @@ std::string http_utils::standardize_url(const std::string& url) {
     return result;
 }
 
+std::string http_utils::generate_random_upload_filename(std::string &directory) {
+    char *filename = NULL;
```

Comment:
`nullptr` … see #247.

Reply:
Done with the rework of this function to not use asprintf any more